org.eclipse.scanning.points/src/org/eclipse/scanning/points/CollatedStepGenerator.java

```diff
@@ -11,11 +11,32 @@
  *******************************************************************************/
 package org.eclipse.scanning.points;
 
-public class CollatedStepGenerator<CollatedStepModel> extends StepGenerator {
+import org.eclipse.scanning.api.ModelValidationException;
+import org.eclipse.scanning.api.points.models.CollatedStepModel;
+import org.eclipse.scanning.api.points.models.StepModel;
+
+public class CollatedStepGenerator extends StepGenerator {
 
 	CollatedStepGenerator() {
 		super();
 		setVisible(false);
 	}
 
+	@Override
+	protected void validateModel() {
+		super.validateModel();
+		if (!(model instanceof CollatedStepModel)) {
+			throw new ModelValidationException("The model must be a " + CollatedStepModel.class.getSimpleName(),
+					model, "offset"); // TODO Not really an offset problem.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)